tests/e2e/tests/test_prometheus.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+
+from e2e.utils.prometheus.setup_prometheus_server import *
+
+from e2e.utils.constants import DEFAULT_USER_NAMESPACE
+from e2e.utils.utils import (
+    rand_name,
+)
+from e2e.utils.config import (
+    metadata,
+    configure_resource_fixture,
+)
+
+from e2e.conftest import region
+
+from e2e.fixtures.cluster import cluster # Needed for kustomize.
+from e2e.fixtures.kustomize import (
+    kustomize, # Needed for port_forward.
+    clone_upstream, # Needed for kustomize.
+    configure_manifests, # Needed for kustomize.
+)
+from e2e.fixtures.clients import (
+    kfp_client,
+    port_forward, # Needed for kfp_client.
+    host, # Needed for kfp_client.
+    client_namespace, # Needed for kfp_client.
+    session_cookie, # Needed for kfp_client.
+    login, # Needed for session_cookie.
+    password, # Needed for session_cookie.
+)
+
+from kfp_server_api.exceptions import ApiException as KFPApiException
+
+GENERIC_KUSTOMIZE_MANIFEST_PATH = "../../deployments/vanilla"
+
+@pytest.fixture(scope="class")
+def kustomize_path():
+    return GENERIC_KUSTOMIZE_MANIFEST_PATH
+
+@pytest.fixture(scope="class")
+def prometheus_amp(request, metadata, region, kustomize):
+    metadata_key = "prometheus-amp"
+    resource_details = {}
+
+    def on_create():
+        cluster_name = metadata.get("cluster_name")
+        print("PROMETHEUS_PRINT: Setting up the workspace id in the fixture.")
+        workspace_id = set_up_prometheus_for_AMP(cluster_name, region)
+        print("PROMETHEUS_PRINT: Setting the workspace_id in the fixture.")
+        resource_details["workspace_id"] = workspace_id
+        print("PROMETHEUS_PRINT: Finished on_create of fixture.")
+        
+    def on_delete():
+        print("PROMETHEUS_PRINT: Entered on-delete function of fixture.")
+        details = metadata.get(metadata_key)
+        cluster_name = metadata.get("cluster_name")
+        delete_AMP_resources(cluster_name, region, details["workspace_id"])
+
+        print("PROMETHEUS_PRINT: Finished on-delete function of fixture.")
+        
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details=resource_details,
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+class TestPrometheus:
+    @pytest.fixture(scope="class")
+    def setup(self, metadata, prometheus_amp):
+        print("PROMETHEUS_PRINT: Starting setup.")
```

Comment:
Overall there are many print statements that give us the status of the test, however I don't see any that print out any of the variables that are being populated. Would any of them make debugging easier if they would be printed out?

Reply:
I agree, and I am doing some prints for debugging and sanity checking in the check_AMP_connects_to_prometheus() function. This includes printing the expected value, the value retrieved from Prometheus, the value retrieved from AMP, and the AMP workspace ID. Can you think of anything else I should add?

Reply:
Not anything esle after reviewing, those all look like the most valuable